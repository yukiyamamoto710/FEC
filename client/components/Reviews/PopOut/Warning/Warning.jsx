import React from 'react';
import PropTypes from 'prop-types';

const Warning = (props) => {
  const { listWarning, isShowWarning } = props;
  return (
    <div
      data-testid="WarningContainer"
      className="warningContainer"
    >
      You must enter the following:
      { listWarning.map((i) => (
        <div
          key={i}
          data-testid="WarningItems"
        >
          <div className="warningItems">
            {i}
          </div>
        </div>
      ))}
      <div>
        <div className="warningButtonBG" />
        <button
          data-testid="WarningButton"
          type="button"
          className="warningButton"
          onClick={isShowWarning}
        >
          Back
        </button>
        <div className="warningButtonPart1" />
        <div className="warningButtonPart2" />
      </div>
    </div>
  );
};

Warning.propTypes = {
  listWarning: PropTypes.arrayOf(PropTypes.string),
  isShowWarning: PropTypes.func,
};

Warning.defaultProps = {
  listWarning: [],
  isShowWarning: () => (1),
};

export default Warning;
