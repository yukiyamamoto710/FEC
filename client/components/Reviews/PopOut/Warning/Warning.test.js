/**
 * @jest-environment jsdom
 */
import React from 'react';
import renderer from 'react-test-renderer';
import {
  render,
  cleanup,
  fireEvent,
  screen,
} from '@testing-library/react';
import Warning from './Warning';

afterEach(cleanup);

const testArray = ['A', 'B', 'C'];
const testFunc = () => { testArray.push('D'); };

it('Waring will render all the warnings', () => {
  const { getAllByTestId } = render(<Warning listWarning={testArray} />);
  const mapArray = getAllByTestId('WarningItems').map((i) => i.innerHTML);
  expect(JSON.stringify(mapArray.length)).toEqual(JSON.stringify(testArray.length));
});

it('Warning will trigger when clicked', () => {
  const handleClick = jest.fn();
  // const { getByTestId } = render(<Warning show={handleClick} />);
  render(<Warning isShowWarning={handleClick} />);
  fireEvent.click(screen.getByTestId('WarningButton'));
  // fireEvent.click(getByTestId('WarningButton'));
  expect(handleClick).toHaveBeenCalledTimes(1);
  // expect(testArray.length).toBe(4);
});

it('should have default isShowWarning()', () => {
  expect(Warning.defaultProps.isShowWarning()).toBeDefined();
});

it('matches snapShot', () => {
  const tree = renderer.create(
    <Warning
      listWarning={testArray}
      isShowWarning={testFunc}
    />,
  )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
