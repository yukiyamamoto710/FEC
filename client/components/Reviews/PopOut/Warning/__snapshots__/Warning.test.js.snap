// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapShot 1`] = `
<div
  className="warningContainer"
  data-testid="WarningContainer"
>
  You must enter the following:
  <div
    data-testid="WarningItems"
  >
    <div
      className="warningItems"
    >
      A
    </div>
  </div>
  <div
    data-testid="WarningItems"
  >
    <div
      className="warningItems"
    >
      B
    </div>
  </div>
  <div
    data-testid="WarningItems"
  >
    <div
      className="warningItems"
    >
      C
    </div>
  </div>
  <div>
    <div
      className="warningButtonBG"
    />
    <button
      className="warningButton"
      data-testid="WarningButton"
      onClick={[Function]}
      type="button"
    >
      Back
    </button>
    <div
      className="warningButtonPart1"
    />
    <div
      className="warningButtonPart2"
    />
  </div>
</div>
`;
